### DISCUSSION TITLE
hurtAndBreak not working properly on curios item

### DISCUSSION THREAD

**Discussion by s1avko** [TheIllusiveC4/Curios#355] [Q&A] [answered]
Hi, 
I am currently working on a ring item to allow creative flight as part of my mod. To apply damage to my Item while flying I am using the hurtAndBreak method provided by the ItemStack class. Damage is being updated every 200 ticks while flying. However, damaging only works extremly inconsistent at random times and I can't figure out why. When it DOES work and the damage gets applied, flying ends immediately without any readable errors. I am using the newest Curios version with Forge 1.20.1

```
@Override
public void curioTick(SlotContext slotContext, ItemStack stack) {

        LivingEntity livingEntity = slotContext.entity();

        if (livingEntity instanceof Player) {
            Player player = (Player) livingEntity;
            player.getAbilities().mayfly = true;    //grant flying
            player.isInvulnerableTo(player.damageSources().fall());
            player.onUpdateAbilities();     //update grant flying
  
            if (player.getAbilities().flying == true) { //check if player is flying and only damage item every 200 ticks when player is flying
                ticksPassed = ticksPassed + 1;
                if (ticksPassed == 200){
                    stack.hurtAndBreak(1, livingEntity, damager -> CuriosApi.broadcastCurioBreakEvent(slotContext)); //damage item and suddenly break
                    System.out.println("DAMAGE: " + stack.getDamageValue() + "/" + stack.getMaxDamage());   
                    ticksPassed = 0;
                }

            }
        }

        ICurioItem.super.curioTick(slotContext, stack);

}
```

When looking at the console while flying you can clearly see the same damage value being printed every 200 ticks and when it does update the damage eventually, it happens at random times and flying ends immediately. Please help

**Comment by TheIllusiveC4** [comment, marked as answer]
Where is your `ticksPassed` being stored? If it's somewhere where both the client and server threads are able to access it, then you may be suffering from a concurrency issue where the client thread is processing information that the server needs before the server can access it.

I'd advise changing your `ticksPassed` logic to instead check against the player's tick count, `player.tickCount % 200 == 0`. In addition, check either `!livingEntity.level().isClientSide()` or `livingEntity instanceof ServerPlayer` to only run it on the server thread.

**Reply by s1avko** [reply]
thank you, checking against player.tickCount worked and solved nearly all issues, after adjusting a little bit more everything works perfectly now. and thanks for the quick reply
